.github/workflows/configure_go.sh

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+#
+# This configures Go according to go.mod, choosing the a GOROOT based on existing variables.
+# For example, if go.mod includes "go 1.17", GOROOT=${GOROOT_1_17} or ${GOROOT_1_17_X64}
+#
+# Variables such as GOROOT_1_17_X64 are defined by GitHub Actions runners. So, this will not
+# result in downloading or installing anything not already there.
+#
+# Notes:
+# * In GitHub, these evaluate to ${RUNNER_TOOL_CACHE}/go/${GO_VERSION}*/x64
+#   * RUNNER_TOOL_CACHE lags Go releases by 1-2 weeks https://github.com/actions/virtual-environments
+# * To simulate GitHub for testing, set GITHUB_ENV and GITHUB_PATH to temporary files
+#   * Ex. `GITHUB_ENV=/tmp/test-env GITHUB_PATH=/tmp/test-path .github/workflows/configure_go.sh
+# * This uses bash because we need indirect variable expansion and GHA runners all have bash.
+set -uex pipefail
+
+go_version=$(sed -n 's/^go //gp' go.mod)
+echo GO_VERSION="${go_version}" >> "${GITHUB_ENV}"
```

Comment:
I would prefer rename this to `GO_MINOR_VERSION` for clarity per https://github.com/tetratelabs/func-e/pull/385#discussion_r720942288